packages/trader/src/Modules/Trading/Components/Form/WheelPicker/wheel-picker-mobile.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useSwipeable } from 'react-swipeable';
+import { Text } from '@deriv/components';
+import { motion } from 'framer-motion';
+
+type WheelPickerMobileProps = {
+    defaultValue?: string;
+    onChange?: (val: string) => void;
+    options: string[];
+    currency: string;
+    optionHeight?: number;
+};
+
+type SwipeState = {
+    startY: number;
+    deltaY: number;
+    translateY: number;
+};
+
+export function getTargetIndex({
+    deltaY,
+    snapTolerance,
+    optionHeight,
+    options,
+    selectedIndex,
+}: {
+    deltaY: number;
+    snapTolerance: number;
+    optionHeight: number;
+    options: string[];
+    selectedIndex: number;
+}) {
+    const absDeltaY = Math.abs(deltaY);
+    const yLeftover = absDeltaY % optionHeight;
+    const snapThreshold = Math.round(optionHeight * snapTolerance);
+    const snapY = Number(yLeftover >= snapThreshold);
+
+    const absDeltaIndex = Math.floor(absDeltaY / optionHeight) + snapY;
+    const deltaIndex = absDeltaIndex * Math.sign(deltaY) * -1;
+
+    const MIN_INDEX = 0;
+    const MAX_INDEX = options.length - 1;
+    const targetIndex = Math.min(MAX_INDEX, Math.max(MIN_INDEX, selectedIndex + deltaIndex));
+    return targetIndex;
+}
+
+const calculateLimits = (swipeStartY: number, deltaY: number, optionHeight: number, optionsLength: number) => {
+    const MIN = 0;
+    const MAX = optionHeight * (optionsLength - 1);
+    const translateY = Math.min(MAX, Math.max(MIN, swipeStartY + deltaY));
+    const newIndex = Math.min(optionsLength - 1, Math.max(0, Math.round(translateY / optionHeight)));
+
+    return { translateY, newIndex };
+};
+
+const WheelPickerMobile: React.FC<WheelPickerMobileProps> = ({
+    defaultValue,
+    onChange,
+    currency,
+    options,
+    optionHeight = 28,
+}) => {
+    const [selectedIndex, setSelectedIndex] = useState<number>(options.length - 1);
+    // const [optionWidth, setOptionWidth] = useState<number>(0);
+
+    const [swipe, setSwipe] = useState<SwipeState>({ startY: 0, deltaY: 0, translateY: 0 });
+    const optionRef = useRef<HTMLDivElement | null>(null);
```

Comment:
I think it might work without null in type, like this: `const optionRef = useRef<HTMLDivElement>(null);`
Noticed just because below you used the same